notebooks/03-query/11-joins.ipynb

```diff
@@ -0,0 +1,2475 @@
+      "  \"start_time\": \"2023-10-24 00:56:37\",\n",
+      "  \"run_time\": 172900.456633,\n",
+      "  \"status\": \"COMPLETE\",\n",
+      "  \"status_msg\": \"Running, Sim Clock: 2023-10-26 00:58:17.457633\"\n",
+      "}\n",
+      "2023-10-26 00:58:30.681189 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 0, 58, 30, 681189)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "wait_for_datagen( \"clicks\")  # this takes about 2 minutes\n",
+    "print_timestamp()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "adbbe00c-2b79-47d4-91df-1d93f942b79f",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "2023-10-26 00:58:30.700274 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "<Response [200]>"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "print_timestamp()\n",
+    "datagen_request = {\n",
+    "    \"name\": \"users\",\n",
+    "    \"target\": { \"type\": \"file\", \"path\":\"users.json\"  },\n",
+    "    \"config_file\": \"clickstream/users_init.json\", \n",
+    "    \"concurrency\":4000,\n",
+    "    \"total_events\":4000 \n",
+    "}\n",
+    "datagen.post(\"/start\", json.dumps(datagen_request), headers=headers)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "2beda33c-6f22-456d-9189-681d547cca8a",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "{\n",
+      "  \"name\": \"users\",\n",
+      "  \"config_file\": \"clickstream/users_init.json\",\n",
+      "  \"target\": {\n",
+      "    \"type\": \"file\",\n",
+      "    \"path\": \"/files/users.json\"\n",
+      "  },\n",
+      "  \"active_sessions\": 0,\n",
+      "  \"total_records\": 4000,\n",
+      "  \"start_time\": \"2023-10-26 00:58:30\",\n",
+      "  \"run_time\": 659.275399,\n",
+      "  \"status\": \"COMPLETE\",\n",
+      "  \"status_msg\": \"Running, Sim Clock: 2023-10-26 01:09:29.980941\"\n",
+      "}\n",
+      "2023-10-26 00:58:31.830364 <<<<<<<<<<<<< \n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "datetime.datetime(2023, 10, 26, 0, 58, 31, 830364)"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "wait_for_datagen( \"users\")\n",
+    "print_timestamp()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ddd8faf9-a267-4607-83a0-85c7720c39cc",
+   "metadata": {},
+   "source": [
+    "### Load `clicks` and `users` Into Separate Tables\n",
+    "\n",
+    "Load separate tables for `clicks` and `users` to demonstrate join funtionality."
```

Comment:
Think you may have to add something here to explain what just happened - are you able to show the raw data anyhow?

Reply:
done. explained where the files are created and provided links to see them.